File: app/src/main/java/com/mahadum360/mahadum/auth/AuthContract.java
package com.mahadum360.mahadum.auth;


import com.mahadum360.mahadum.models.CreateUser;
import com.mahadum360.mahadum.models.LoginUser;
import com.mahadum360.mahadum.models.User;

public interface AuthContract {

    interface View {
        void hideProgress();

        void showProgress();

        void showComplete();

        void showError(String call, String message);

        void onLoginSuccess();

        void onRegistrationSuccess(User user);

        void onLogout();
    }

    interface Presenter {
        void logout();

        void login(LoginUser user);

        void register(CreateUser user);

        void detachView();
    }
}

File: app/src/main/java/com/mahadum360/mahadum/auth/AuthContract.kt
package com.mahadum360.mahadum.auth


import com.mahadum360.mahadum.models.*

interface AuthContract {

    interface View {
        fun hideProgress()
        fun showProgress()
        fun showComplete()
        fun showError(call: String, message: String)
        fun onLoginSuccess(user: User)
        fun onRegistrationSuccess(user: User)
        fun onValidatePasswordSuccess(response: String)
        fun onChangePasswordSuccess(response: String)
        fun onLogout()
    }

    interface Presenter {
        fun logout()
        fun login(user: LoginUser)
        fun register(user: CreateUser)
        fun validatePassword(password: PasswordRequest)
        fun changePassword(changePassword: ChangePassword)
        fun detachView()
    }
}
